### PR TITLE
Implement timeline

### PR DESCRIPTION
**Pull request by pahor167**
image
image
image

## Summary
- Added a vertical timeline to the proposal detail page showing lifecycle phases (Upvoting → Voting → Execution) and key events (Approved, Rejected, etc.)
- Each step has a colored status dot: green for completed, purple pulsing for active, grey for future, red for failed
- Timestamps shown in local timezone with UTC on hover
- Active phases show a countdown ("Ends in 3 days")
- Handles edge cases like rejected proposals, missed approvals, withdrawn proposals, and missing DB timestamps (back-calculates from available data)
- Replaces the old single-line "Voting ends in..." text

## Test plan
- [x] Unit tests for `buildTimelineSteps` covering all proposal states (26 tests passing)
- [x] Manually tested with real proposals: active voting, execution, executed, rejected, expired, withdrawn
- [x] Prettier and lint checks pass

### CODE REVIEW

File: src/features/governance/components/TIMELINE.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+```
+MergedProposalData (queuedAt, dequeuedAt, approvedAt, executedAt, stage, transactionCount)
+        |
+        v
+buildTimelineSteps() — pure function, computes steps from proposal data + quorumMet
+        |
+        v
+TimelineStep[] — rendered as a vertical stepper
+```
```
[the remark this answers is not in the document]
Comment (pahor167) [reply src/features/governance/components/TIMELINE.md]: mermaid added
